--- /usr/lib/python2.7/site-packages/cloudinit/distros/__init__.py.ORIG 2014-04-01 20:26:07.000000000 +0200
+++ /usr/lib/python2.7/site-packages/cloudinit/distros/__init__.py      2016-12-07 11:20:10.580797643 +0100
@@ -85,7 +85,7 @@
     def set_hostname(self, hostname, fqdn=None):
         writeable_hostname = self._select_hostname(hostname, fqdn)
         self._write_hostname(writeable_hostname, self.hostname_conf_fn)
-        self._apply_hostname(hostname)
+        self._apply_hostname(writeable_hostname)
 
     @abc.abstractmethod
     def package_command(self, cmd, args=None, pkgs=None):
